src/NServiceBus.Core/Encryption/RijndaelEncryptionService.cs

```diff
@@ -29,54 +29,75 @@ namespace NServiceBus.Encryption.Rijndael
     using System;
     using System.Collections.Generic;
     using System.IO;
-    using System.Linq;
     using System.Security.Cryptography;
-    using System.Text;
+    using NServiceBus.Logging;
 
     class RijndaelEncryptionService : IEncryptionService
     {
-
+        static readonly ILog Log = LogManager.GetLogger<RijndaelEncryptionService>();
+        readonly IBus bus;
+        readonly string encryptionKeyIdentifier;
         byte[] encryptionKey;
-        List<byte[]> decryptionKeys;
-
-        public RijndaelEncryptionService(string encryptionKey, List<string> expiredKeys)
+        IDictionary<string, byte[]> keys;
+        IList<byte[]> decryptionKeys; // Required, as we decrypt in the configured order.
+
+        public RijndaelEncryptionService(
```

Comment:
@SimonCropp @indualagarsamy What do you think of this? I don't like this dependency of objects and configuration but don't know how to improve it. I read something about abstract factory but I don't know if this should be applied at this implementation.